pkg/progressreader/progressreader.go

```diff
@@ -0,0 +1,46 @@
+package progressreader
+
+import (
+	"github.com/docker/docker/utils"
+	"io"
+)
+
+// Reader with progress bar
+type Config struct {
+	In         io.ReadCloser // Stream to read from
+	Out        io.Writer     // Where to send progress bar to
+	Formatter  *utils.StreamFormatter
```

Comment:
You will have to create an interface within this package that has the same method signatures as the type so that we don't have an import into the utils package.